src/services/organizeImports.ts

```diff
@@ -162,7 +163,7 @@ namespace ts.OrganizeImports {
 
         function isDeclarationUsed(identifier: Identifier) {
             // The JSX factory symbol is always used if JSX elements are present - even if they are not allowed.
-            return jsxElementsPresent && (identifier.text === jsxNamespace || jsxFragmentFactory && identifier.text === jsxFragmentFactory) ||
+            return jsxElementsPresent && (identifier.text === jsxNamespace || jsxFragmentFactory && identifier.text === jsxFragmentFactory) && !(compilerOptions.jsx === JsxEmit.ReactJSX || compilerOptions.jsx === JsxEmit.ReactJSXDev) ||

```

Comment:
In `importFixes.ts`, I have a very small utility called `jsxModeNeedsExplicitImport`. Do you want to move that to `utilities.ts` and use it here so in the future we fix more of these things at one time if we need to account for new modes?

Reply:
@andrewbranch Do I need to extend it with ReactJSX and ReactJSXDev?

Reply:
No, my utility says which ones _need_ an import, which is _only_ React and ReactNative. We do not require an import for Preserve, so it does have different behavior than what you have now.

Reply:
Oke, I've pushed changes. Thanks